Set up authentification for flask api #94

Eik-S opened the pull request:
Resolves #94 

**Merge with `change-api-calls-to-post-requests-#121` at tracemap-backend**

# Important changes

- changed flask routes to post requests in `server.py`
- **offtopic:** fixed bug where `get_tweet_info` route was throwing an error for invalid tweet ids, now its returning an empty response (`None`)

# Review hints

Please have a look in the according **tracemap-tool pull request** tracemap/tracemap-tool#122 for the review hints and review both requests at the same time.

jonathanschlue commented:
And you added token authentication, which is actually securing the api requests.

jonathanschlue commented on server.py:
```diff
@@ -13,54 +13,106 @@
+def __is_session_valid(email, session_token):
+    if email and session_token:
+        response = betaAuth.check_session(email, session_token)
+        if response['session'] == True:
```
If providing an invalid (here spoiled) session token, `response['session']` throws a `KeyError`, which leads to a `500` server response:

![bildschirmfoto vom 2018-11-01 18-12-30](https://user-images.githubusercontent.com/15818147/47867430-64a61000-ddf9-11e8-8630-d6dbc31e8240.png)

![bildschirmfoto vom 2018-11-01 18-11-44](https://user-images.githubusercontent.com/15818147/47867436-67086a00-ddf9-11e8-8b11-1efe756ab37f.png)

The problematic code for this might reside in `beatAuth.check_session`.

jonathanschlue commented on server.py:
```diff
@@ -13,54 +13,106 @@
+    body = request.get_json()
+    session_token = body['session_token']
+    email = body['email']
+    tweet_id = body['tweet_id']
```
Lines 38-40 potentially lead to a lot of `KeyErrors` when any of these fields are not set in json:

![bildschirmfoto vom 2018-11-01 18-36-48](https://user-images.githubusercontent.com/15818147/47868676-c025cd00-ddfc-11e8-8700-c3b7c7920e6e.png)

I suggest using something like 

 ```python
    session_token = body.get('session_token', None)
    email = body.get('email', None)
    tweet_id = body.get('tweet_id', None)
```

Then, if any required parameter is missing (`None`) a `Response("Bad Request", status=400)` should be returned.

Eik-S commented:
I fixed the key error for checking the `session_token` and added a check for the body.
If a body parameter is now missing, the request returns `400 - Bad Request`.

jonathanschlue commented on server.py:
```diff
@@ -16,7 +16,7 @@
 def __is_session_valid(email, session_token):
```
I also suggest that we introduce type hints for readability and understanding, e.g.:

```python
def __is_session_valid(email: str, session_token: str) -> bool:
```
